spec/jobs/reindex_model_job_spec.rb

```diff
@@ -24,11 +33,39 @@
       block&.call
     end
 
-    Apartment::Tenant.switch!(account.tenant) { work }
+    Apartment::Tenant.switch!(account.tenant) do
+      work
+      private_work
+      pending_review_work
+    end
+  end
+
+  context "mint doi" do
+    let(:sipity_entity) { instance_double(Sipity::Entity, proxy_for_global_id: "gid://hyku/#{work.class}/#{work.id}", workflow_id: workflow.id, workflow_state: sipity_workflow_state) }
+
+    it "when work visibility is public & doi is not present" do
+      allow(Sipity::Entity).to receive(:find_by) { object }
+      described_class.perform_now(work.class.to_s, account.cname, limit: 1, options: options)
+      expect(work.reload.doi_status_when_public).to eq "findable"
+    end
   end
 
-  it "sets doi_status_when_public to findable" do
-    described_class.perform_now(work.class.to_s, account.cname, limit: 1, options: options)
-    expect(work.reload.doi_status_when_public).to eq "findable"
+  context "skip doi minting" do
+    let(:sipity_entity) { instance_double(Sipity::Entity, proxy_for_global_id: "gid://hyku/#{pending_review_work.class}/#{pending_review_work.id}", workflow_id: workflow.id, workflow_state: sipity_workflow_state) }
```

Comment:
Could we shorten these longer lines, they're getting kind of gnarly, this one and L44. 

```
let(:gid) { "gid://hyku/#{pending_review_work.class}/#{pending_review_work.id}" }
let(:sipity_entity) { instance_double(Sipity::Entity, proxy_for_global_id: gid, workflow_id: workflow.id, workflow_state: sipity_workflow_state) }
```

Reply:
Thanks I will :+1: